VENDORING.md

```diff
@@ -0,0 +1,238 @@
+developer.
+`Gopkg.toml` is automatically generated by `dep init`, and can be modified
+in order to constraint any package to follow a specific revision.
+
+Here is what `Gopkg.toml` should look like:
+```toml
+# Gopkg.toml example
+#
+# Refer to https://github.com/golang/dep/blob/master/docs/Gopkg.toml.md
+# for detailed Gopkg.toml documentation.
+#
+# required = ["github.com/user/thing/cmd/thing"]
+# ignored = ["github.com/user/project/pkgX", "bitbucket.org/user/project/pkgA/pkgY"]
+#
+# [[constraint]]
+#   name = "github.com/user/project"
+#   version = "1.0.0"
+#
+# [[constraint]]
+#   name = "github.com/user/project2"
+#   branch = "dev"
+#   source = "github.com/myfork/project2"
+#
+# [[override]]
+#  name = "github.com/x/y"
+#  version = "2.4.0"
+
+```
+This file is interesting because it contains all the information the developer
+needs to modify it properly.
+
+From this point, you can vendor any package needed from your code.
+
+### Vendor a newly imported package
+
+This case describes a situation where the developer is adding some new code
+relying on a package never imported before. Let's say the new package is
+`github.com/foo/bar`, here is what needs to be done:
+```bash
+dep ensure
+dep prune
+```
+
+At this point, only the needed files will have been copied into `vendor`.
+
+One more thing we might want here is to ensure the vendored package refers
+to a specific version. This is what is called a constraint with `dep`.
+By modifying `Gopkg.toml`, the developer can decide to force our package
+`github.com/foo/bar` to point to the revision `f5742cb6`. Here is what the
+file should look like:
+```toml
+[[constraint]]
+  name = "github.com/foo/bar"
+  revision = "f5742cb6"
+```
+
+Or in case he wants to point to a specific version `1.2.4`:
+```toml
+[[constraint]]
+  name = "github.com/foo/bar"
+  version = "1.2.4"
+```
+
+Or in case he wants to point to a specific branch `work-dev`:
+```toml
+[[constraint]]
+  name = "github.com/foo/bar"
+  branch = "work-dev"
+```
+
+At last, a very powerful constraint is to be able to specify if the source
+of repository comes from a forked repository:
+```toml
+[[constraint]]
+  name = "github.com/foo/bar"
+  branch = "fork-work-dev"
+  source = "github.com/myfork/bar"
+```
+
+__Note:__ Constraint a package is important because this is the only way to
+make sure about the version of an imported package. This prevents from any
+build failure that could result from a package change on the upstream.
+
+### Update a vendored package
+
+This case describes a situation where the developer adds some code relying on
+another version of a package already vendored. This is a very simple case since
+the package is already part of the vendoring. The developer will have to modify
+`Gopkg.toml` by updating the vendoring version:
+
+_Original file_
+```toml
+[[constraint]]
+  name = "github.com/foo/bar"
+  revision = "f5742cb6"
+```
+
+_Updated file_
```

Comment:
This is great, but as this doc is for developers, I'd be inclined to just show the diff. You then get to use the funky `` ```diff ``  command which shows colourised output:

```diff
$ diff -u Gopkg.toml.orig Gopkg.toml.new
--- Gopkg.toml.orig     2018-01-23 08:42:03.738664995 +0000
+++ Gopkg.toml.new      2018-01-23 08:42:12.036598793 +0000
@@ -1,3 +1,3 @@
 [[constraint]]
   name = "github.com/foo/bar"
-  revision = "f5742cb6"
+  revision = "a4be45c7"
```

If you agree, could you apply that logic to all the original+updated pairs below?

Reply:
Oh yeah I like it, it's cleaner !

Reply:
done